src/convertToObject.js

```diff
@@ -5,8 +5,26 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(stylesString) {
+  return stylesString
+    .split(';')
+    .map((style) => style.trim())
+    .filter(Boolean)
+    .map((style) => {
+      const [property, value] = style.split(':');
+
+      if (property === undefined || value === undefined) {
+        return null;
+      }
```

Comment:
[CODE KNOWLEDGE] - The check for `undefined` properties and values is unnecessary here. The `split` method on a string will always return an array with at least one element, even if the delimiter is not found. If a style declaration is malformed and doesn't contain a colon, the `split` will return an array with the entire string as the first element, and `undefined` as the second. However, since you are already using `.filter(Boolean)` to remove falsy values, this check will never be true because a string with no colon will not produce a falsy value after the split. You can safely remove this check.